app/src/main/java/org/simple/clinic/security/pin/PinEntryUpdate.kt

```diff
@@ -14,16 +14,26 @@ class PinEntryUpdate(
 
   override fun update(model: PinEntryModel, event: PinEntryEvent): Next<PinEntryModel, PinEntryEffect> {
     return when (event) {
-      is PinTextChanged -> next(model.enteredPinChanged(event.pin))
-      is PinDigestToVerify -> next(model.updatePinDigest(event.pinDigest))
+      is PinTextChanged -> {
+        val updatedModel = model.enteredPinChanged(event.pin)
+
+        next(updatedModel, generateEffectsForPinSubmission(updatedModel))
+      }
+      is PinDigestToVerify -> {
+        val updatedModel = model.updatePinDigest(event.pinDigest)
+
+        next(updatedModel, generateEffectsForPinSubmission(updatedModel))
+      }
       is PinEntryStateChanged -> {
         val effects = when (val protectedState = event.state) {
           is Allowed -> setOf(generateEffectForAllowingPinEntry(protectedState), AllowPinEntry)
           is Blocked -> setOf(ShowIncorrectPinLimitReachedError(protectedState.attemptsMade), BlockPinEntryUntil(protectedState.blockedTill))
         }
 
-        dispatch(effects)
+        Next.dispatch(effects)
```

Comment:
Nice! I didn't know we can dispatch a list of `Effect`s.

Reply:
Yup, this is actually the default version provided by Mobius. The vararg version is a custom top-level function we wrote for convenience.